### ISSUE TITLE
Setting a default columnSize value for dash_ag_grid

### ISSUE THREAD

**Issue by petar-qb** [mckinsey/vizro#681]
Reference link: https://dash.plotly.com/dash-ag-grid/column-sizing#size-to-fit-and-responsive-size-to-fit

Should Vizro set `columnSize="responsiveSizeToFit"` for the `dash_ag_grid` figure function by default?

**Comment by huong-li-nguyen** [comment]
@maxschulz-COL and I once discussed this several times 😄 I actually don't know what we ended up with.

So this is how I would expect it to work:
- The Grid always takes on the full width of the screen size so even if there are only 3 columns, the width of the grid will be the entire page (as in the dash_data_table)
- If there are more columns in the Grid that it fits in width, a scroll bar will appear horizontally
- All columns are always well visible and not squeezed (so should be fully readable)

Now, I don't know what configuration achieves this (if any?)

**Comment by maxschulz-COL** [comment]
Mostly agree with the above. But I would note a few things:
- having columns and their content fully readable can also be annoying when there are outliers
- I remember that problems also occurred also when pre-setting things, but this issue may have been removed now with your work and providing just a Div, and not an object

**Comment by huong-li-nguyen** [comment]
I quickly discussed this with Joe and from a design perspective we would prefer that it expands to its full width (which it currently does not do). However, there are pros/cons for both approaches and the preference is not significant. So we decided to not change anything for now and let users configure it themselves 👍 

This means for our demos, please set `columnSize="responsiveSizeToFit"` manually for now if the table does not take the full width :)

**Comment by antonymilne** [comment]
With `vm.Flex` it's not possible to directly get AgGrid to expand to full width: [pycafe example](https://py.cafe/snippet/vizro/v1?#c=H4sIAM-H9mcEA41VTW_jNhD9K4J6cQCblS1_Ayqw7WK3x0UP7WEVBLREWVxTpEpSSeTA_71vKDuxgQbZ5JBw5s28Nx-kXuLClCLexrJpjfXRozxawxrYlIu4ix6bXN-4WmW86pl4bq1wAdI-57qyphkA0Rn9Nx2uHczznRLu4i-5qx_4_mFvZZnrXJdVlCEVK7nnbM_bRupS2NEd-Vq-F_A-Nuwb_hvlOsKPl16JLI8_i4p3ykefkTD69DX6Sgnj8QBSvDedzxD5RYnn0d3ZXBiI0EJ7l32H79OegkayRLqgJx5Hldx3VmTXMkek7aGyvIG9GkfO91DwkqOFytg83kZ5bAWiT3d39-Mo10E8ZdgZbsuhApIZjiOqCvz05z4gQ8tGd2zXSUVkZ-Ads50eGhGPQfBvJ61oSDzGFlr7W5awKUtXAeD7luYZHDi2va-NhqHtTSlLMXlM2GzJpnANzYm3L_GjsE4SagYCY_xf5onM51QWp3Fc1FBlBUDfXz0YqRMezidZ-jreThfJOMbk_hmO6XD6U8h9DR46yhJhlVTid2R1wv5htOdSC_sOA0EnuwELSMspbxyf7k_jV8ybijPResaWy818tU7T-WqTJMv_z424t5S8bVnbx7dpr9wOFO7XonPeNKxw7laCDS268E_TKVsm6Xq1TpbLxWaezN8VMAj_uGXnFnzUrkunbsS90tQXfavFz3B6YWHm6iPSC45Y6fc0DmPDegKjcVuAOfePiab1_RejcLe_KV6IOvxLiUGCnQY0zOA27Krt18hf6FZGL9ET1bGNOg1wdIIM7LYpDiSCtJ_BUY0btY0Enxez6booV8myXKSbUuzELi2SZLMUfDercs21Np57UU6oQJfhdi1ZAnvnTaX4QWTZjKVsmusdSjjgtMD1A2CnpD4Im2VTtqFzwYtaKLmjDNM0WIT1spIISWZzNmMzmGpuoXuijW24kkeKT5GeXEpS-jWy427Tg0DM0zX56DTh-wm1AAG4_iQoWHe4wc5b3k6u3jloWr0iCmPFjXPGEpIXnLVv1LvOhmuPVbjxo7gFS88ALBsexWwxxKBdDhWgdgKE04S6IvWe8gbNstScIChx-DagY5NGeE7PLVWP1wou70qu98KaLkiCqFz_kPoHn1Ew6s91w-2hax2vwoSmbAFT3_YT8YwdoAcOkRgURWpun2qugiGdE4EWzk-463UhDVnDyHXXtD2d8GTOc3yIqMhXenobXCuK0AIQwqK4rzDIUlrA5igQVQ2fzCxDxoDpUYiXiJphJJTmbEBT7aAcEydzWDbkOa_D5QWvvW-JcmAM7_sEvRKdl4rCsXusNc6H1P4IEy0bsPTtQJWhAGwM_FZ424dh0AKhQmxi13pjqDPLDcNiwyafyT8NHcGVAP6mpcgdkvnjMLELXWcVRpkSW5AaPkmhV-Fb9STs4Sg6cNN6LGHAyvo3pqNsUSYqD8mdU7mOT_8BEx5_tbAIAAA).

Note the `style={"color": "red"}` applies to the inner `Grid.figure` (with `id=self._input_component_id`) but what we actually need to target is the outer `html.Div(id=self.id)`. This is possible if you assign an `id` to the outer `vm.AgGrid` and then target it with CSS in a stylesheet, but that's a bit annoying.

It doesn't look like there's an easy way to move these IDs around so might need to rethink how we do CSS here: https://github.com/mckinsey/vizro/blob/c1e80c2240c2e08b982f9f9319309abea14dc45f/vizro-core/src/vizro/static/css/flex.css#L9.

**Comment by huong-li-nguyen** [comment]
With a fixed width for the AgGrid inside `flex`, there are always going to be cases where it doesn’t quite work. 😬  

Right now, the dimensions are based on Plotly figures, so the behavior stays consistent. I think the width is already a bit bigger than usual. And since Plotly figures don’t adjust their logic for rows or columns, I feel like keeping the same approach for the table works ok for now.  

That said, I think the best solution would be for the `table-container` to automatically adjust its width based on the table’s actual content, no matter if it’s in a `flex-row` or `flex-column`. The `ag-header-row` has the relevant number in its `style` attribute, so we’d probably need to use JavaScript to dynamically set the width and make sure it works across all the nested setups. However, I don't know how complicated it is to implement this, but might be worth exploring it in the next sprint if the fixed width doesn't work for us? So up to you guys and how much capacity you have :) 

![Screenshot 2025-04-11 at 12 39 16](https://github.com/user-attachments/assets/4adf24d1-5265-45c9-a462-f22523c86d6e)

**Comment by huong-li-nguyen** [comment]
Please check how it works now, and I'll close this issue after 👍

**Comment by petar-qb** [comment]
I really like this change!

There's just one case that confuses me. While AgGrid stretches properly in the horizontal way, there are some height inconsistencies with the Table. Should those two be aligned? 

Three vm.AgGrid objects in flex row look like:
<img width="1720" alt="Image" src="https://github.com/user-attachments/assets/ed24db9f-d0e8-4c08-8c07-a6d1a490d900" />

Three vm.Table objects in flex row look like:
<img width="1720" alt="Image" src="https://github.com/user-attachments/assets/6d188adb-ffe9-4b74-919c-52f0eaa16455" />

**Comment by huong-li-nguyen** [comment]
I don't think we need to align these since we're planning to deprecate the `Table` anyway. But just to clarify, both behaviors are technically correct.

In the `Flex` layout, components default to their natural height and width unless we explicitly override them. AgGrid and Plotly charts have their own dimensions set by the relevant libarties, so they display as expected. The `Table`, on the other hand, doesn’t seem to have fixed dimensions, so it ends up expanding to 100%.

In the `Flex` layout, we're actually overriding less— components just take their natural size. In contrast, the `Grid` layout forces all components to fill their grid space, which means 100% width and height. So from that perspective, `AgGrid` behaves correctly in both layouts now. In `Grid`, it’s stretched to fit the cell, and in `Flex`, it just uses its natural size.

**Comment by huong-li-nguyen** [comment]
Closing as this has been implemented here: https://github.com/mckinsey/vizro/pull/1162
